### PR TITLE
Remove support for python 2

### PR DESCRIPTION
**Pull request by jacalata**

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+  pull_request:
+    branches: *
```
Comment (ManishMuttreja1) [inline .github/workflows/python-package.yml:11]: Minor - shouldn't this be `'*'` for satisfying yaml syntax?

File: test/test_xfile.py [hunk 2]
```diff
@@ -28,24 +27,24 @@ def assertContainsUserNamespace(self, filename):
...
 
-'''            
+
+'''
     def demo_bug_ns_not_preserved_if_not_used(self):
         filename = TABLEAU_10_TDS
         self.assertContainsUserNamespace(filename)
         wb = Datasource.from_file(filename)
-        #wb.save() 
+        #wb.save()
```
Comment (ManishMuttreja1) [inline test/test_xfile.py:47]: Why did we add this comment? I hope not a typo.

Comment (jacalata) [reply test/test_xfile.py:47]: If you mean this line, I was checking variations of save and save-as to make sure they worked the same. It can be removed altogether.

The test case as a whole documents an edge case I found which wasn't worth delaying the main fix for, but is a theoretical bug in the future. The test case is kept as documentation, but commented out so the pipelines succeed.

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (jacalata) [reply .github/workflows/python-package.yml:11]: that might explain why the workflow failed :)

File: test/test_xfile.py (hunk 2 above)
Comment (ManishMuttreja1) [reply test/test_xfile.py:47]: Ok thanks, then some more comments in this function would be nice.

**Review by ManishMuttreja1 — APPROVED** [review]
